ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -0,0 +1,291 @@
+            }
+            bid.CurrentOffers.Add(new SellerOfferEntity {
+                BidId = bid.Id,
+                SellerId = supplierProposal.SupplierId,
+                PublishedTime = supplierProposal.PublishedTime,
+                MinimumUnits = supplierProposal.MinimumUnits,
+                OfferedPrice = supplierProposal.ProposedPrice,
+                OfferDescription = supplierProposal.Description,
+            });
+            bid.PotenialSuplliersCounter += 1;
+            using (var new_context = new YotyContext())
+            {
+                try
+                {
+                    new_context.Bids.Update(bid);
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            //TODO dropping the dto obj from the response will save us the second db access 
+            var supplier_ent = await _context.Sellers.FindAsync(supplierProposal.SupplierId).ConfigureAwait(false);
+            //TODO map ent to dto
+            return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response<BidDTO>> CreateNewBid(NewBidRequst productBidRequest)
+        {
+            // TODO use mapper instead
+            ProductEntity product_ent = new ProductEntity() {
+                Description = null,
+                Id = null,
+                Image = null,
+                Name = null,
+            };
+            BidEntity newBid = new BidEntity() {
+                OwnerId = productBidRequest.OwnerId,
+                Category = productBidRequest.Category,
+                SubCategory = productBidRequest.SubCategory,
+                MaxPrice = productBidRequest.MaxPrice,
+                ExpirationDate = productBidRequest.ExpirationDate,
+                Product = product_ent,
+                //TODO is this the time we want? (or global).
+                CreationDate = DateTime.Now,
+                CurrentOffers = new List<SellerOfferEntity>(),
+                CurrentParticipancies = new List<ParticipancyEntity>(),
+                UnitsCounter = 0,
+                Id = this.GenerateBidId().ToString(),
+                PotenialSuplliersCounter = 0,
+            };
+
+            using (var new_context = new YotyContext())
+            {
+                new_context.Bids.Add(newBid);
+                try
+                {
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
```

Comment:
we failed to create a new bid in our db,
the failure message it's an external message we expose to the client - the details on why its happened should stay with us ( we will log them when we will have logger) ... change the message to something like "Failed to create new bid"